hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/BulkUploadHandler.java

```diff
@@ -203,46 +208,48 @@ private void readFileStream(final InputStream tempStream) {
-            resetCounts();
+            resetSuccessfullTargetCount();
         }
 
-        private void syncCountAfterUpload(final long totalNumberOfLines) {
-            if ((successfullTargetCount + failedTargetCount) != totalNumberOfLines) {
-                // something went wrong due to runtimeexception etc.
-                final long syncedFailedTargetCount = totalNumberOfLines - successfullTargetCount;
-                managementUIState.getTargetTableFilters().getBulkUpload()
-                        .setSucessfulUploadCount(successfullTargetCount);
-                eventBus.publish(this, new TargetTableEvent(TargetComponentEvent.BULK_TARGET_CREATED));
-                managementUIState.getTargetTableFilters().getBulkUpload().setFailedUploadCount(syncedFailedTargetCount);
-                managementUIState.getTargetTableFilters().getBulkUpload().setProgressBarCurrentValue(1);
-            }
+        private void syncCountAfterUpload(final int totalNumberOfLines) {
+            final int syncedFailedTargetCount = totalNumberOfLines - successfullTargetCount;
+            managementUIState.getTargetTableFilters().getBulkUpload().setSucessfulUploadCount(successfullTargetCount);
+            managementUIState.getTargetTableFilters().getBulkUpload().setFailedUploadCount(syncedFailedTargetCount);
+            managementUIState.getTargetTableFilters().getBulkUpload().setProgressBarCurrentValue(1);
+            eventBus.publish(this, new TargetTableEvent(TargetComponentEvent.BULK_TARGET_CREATED));
         }
 
-        private long getTotalNumberOfLines() {
+        private BigDecimal getTotalNumberOfLines() {
+            try {
+                return new BigDecimal(Files.readLines(tempFile, Charset.defaultCharset(), new LineProcessor<Integer>() {
+                    private int count;
+
+                    @Override
+                    public Integer getResult() {
+                        return count;
+                    }
+
+                    @Override
+                    public boolean processLine(final String line) {
+                        count++;
+                        return true;
+                    }
+                }));
 
-            long totalFileSize = 0;
-            try (InputStreamReader inputStreamReader = new InputStreamReader(new FileInputStream(tempFile),
-                    Charset.defaultCharset())) {
-                try (BufferedReader readerForSize = new BufferedReader(inputStreamReader)) {
-                    totalFileSize = readerForSize.lines().count();
-                }
-            } catch (final FileNotFoundException e) {
-                LOG.error("Error reading file {}", tempFile.getName(), e);
             } catch (final IOException e) {
-                LOG.error("Error while closing reader of file {}", tempFile.getName(), e);
+                LOG.error("Error while reading of file {}", tempFile.getName(), e);
```

Comment:
You are using a wrong method call. Instead of the e you should provide a second argument not the exception.